src/auth/errors.ts

```diff
@@ -21,3 +24,22 @@ export function isJwtError(error: GraphQLError): boolean {
 export function isTokenExpired(error: GraphQLError): boolean {
   return error.extensions.exception.code === JWTError.expired;
 }
+
+export function getAuthErrorType(graphQLError: GraphQLError): UserContextError {
+  switch (graphQLError.extensions?.exception?.code) {
+    case "PermissionDenied":
+      return "noPermissionsError";
+    case "OAuthError":
+      return "externalLoginError";
+    default:
+      return "unknownLoginError";
+  }
+}
+
```

Comment:
suggestion: please try to avoid such statements that are untyped and use them as strings. It breaks DRY rule a bit as you repeat strings of error cases many times in the code. You can create a constant object (enum-kind equivalent) and refer to the error cases by this, then you have these in place, and you can have "exhaustiveness" in the switch

Reply:
Sure, I changed it, so now `code` is marked as `AuthError` type.

Reply:
Also I changed to return strings as `UserContextError` object member values.